desktop/main.js

```diff
@@ -597,6 +598,16 @@ const mainWindow = () => {
                     }
                 });
 
+                const downloadQueue = createDownloadQueue();
+
+                ipcMain.on(ELECTRON_EVENTS.DOWNLOAD, (event, downloadData) => {
+                    const downloadItem = {
+                        ...downloadData,
+                        win: browserWindow,
+                    };
+                    downloadQueue.pushDownloadItem(downloadItem);
```

Comment:
With a simple queue, you can do the following:

```
queue.push(fn)
```

Reply:
@allroundexperts Which queue implementation were you referring to?

Reply:
Electron-specific code and generic queue are now separated